sympy/core/add.py

```diff
@@ -870,10 +870,51 @@ def _eval_as_leading_term(self, x):
             return expr.as_leading_term(x)
 
         infinite = [t for t in expr.args if t.is_infinite]
+        compute = False
 
-        expr = expr.func(*[t.as_leading_term(x) for t in expr.args]).removeO()
-        if not expr:
-            # simple leading term analysis gave us 0 but we have to send
+        leading_terms = [t.as_leading_term(x) for t in expr.args]
+
+        try:
+            min = Order(leading_terms[0]).getn()
```

Comment:
We have to tell `Order` that `x` is the relevant variable: `Order(leading_terms[0], x).getn()`.